src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,243 @@
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+    setInputValues({
+      title: false,
+      imgUrl: false,
+      imdbUrl: false,
+      imdbId: false,
+    });
+    setCount(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (
+      movie.title
+      && movie.imgUrl
+      && movie.imdbUrl
+      && movie.imdbId
+    ) {
+      onAdd(movie);
+      setCount(count + 1);
+      reset();
+    }
+  };
+
+  const handleOnBlur = (event: React.FocusEvent<HTMLInputElement, Element>) => {
+    const { name, value } = event.target;
+
+    switch (name) {
+      case Input.TITLE:
+        if (!movie.title.trim()) {
+          setInputValues({
+            ...inputValues,
+            title: true,
+          });
+        }
+
+        break;
+
+      case Input.IMGURL:
+        if (!movie.imgUrl || !pattern.test(value)) {
+          setInputValues({
+            ...inputValues,
+            imgUrl: true,
+          });
+
+          setIsImgUrlValid(true);
+        }
+
+        break;
+
+      case Input.IMDBURL:
+        if (!movie.imdbUrl || !pattern.test(value)) {
+          setInputValues({
+            ...inputValues,
+            imdbUrl: true,
+          });
+
+          setIsImdbUrlValid(true);
+        }
+
+        break;
+
+      case Input.IMDBID:
+        if (!movie.imdbId.trim()) {
+          setInputValues({
+            ...inputValues,
+            imdbId: true,
+          });
+        }
+
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const handleChange = (event: ChangeEvent) => {
+    const { name, value } = event.target;
+
+    setMovie(prevState => ({
+      ...prevState,
+      [name]: value,
+    }));
+
+    switch (name) {
+      case Input.TITLE:
+        if (name === Input.TITLE && name === Input.TITLE.trim()) {
+          setInputValues({
+            ...inputValues,
+            title: false,
+          });
+        }
+
+        break;
+
+      case Input.IMGURL:
+        if (name === Input.IMGURL) {
+          setInputValues({
+            ...inputValues,
+            imgUrl: false,
+          });
+
+          setIsImgUrlValid(false);
+        }
+
+        break;
+
+      case Input.IMDBURL:
+        if (name === Input.IMDBURL) {
+          setInputValues({
+            ...inputValues,
+            imdbUrl: false,
+          });
+
+          setIsImdbUrlValid(false);
+        }
+
+        break;
+
+      case Input.IMDBID:
+        if (name === Input.IMDBID) {
+          setInputValues({
+            ...inputValues,
+            imdbId: false,
+          });
+        }
+
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const notActiveButton = (
+    !movie.title
+    || !movie.imdbId
+    || isImgUrlValid
+    || isImdbUrlValid
+  );
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onChange={handleSubmit}
```

Comment:
```suggestion
      onSubmit={handleSubmit}
```
зараз фільм додається коли вводиш всі поля, навіть без кнопки

Reply:
Дякую за допомогу, шукав в чому помилка цілий день)